### PR TITLE
feat(test): add backend integration test

### PR DESCRIPTION
**Pull request by yhwang**
Add backend integration test to the GitHub workflow. it builds backend images, deploys, and runs a simple flip coin test.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by yhwang** [comment]
> For the kind test, can we only trigger it when certain repo files are changed?

Yes, good idea. I also think about can we trigger it by a command you can put in the comments

**Comment by yhwang** [comment]
add code to check if there is any change on the backend. then trigger the integration test if any

File: .github/workflows/kfp-tekton-unittests.yml [hunk 1]
```diff
@@ -109,3 +109,45 @@ jobs:
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: backend changes
+        id: backend-changes
+        run: |
+          git fetch
+          case "${{ github.event_name }}" in
+            pull_request)
+              echo "::set-output name=all::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | grep -e '^backend/' | grep -v '^backend/.*README.md' | xargs)"
```
Comment (Tomcli) [inline .github/workflows/kfp-tekton-unittests.yml:123]: can you also check the go.mod file?

Comment (yhwang) [reply .github/workflows/kfp-tekton-unittests.yml:123]: sure, see if I can find an action to filter change lists. that will be cleaner

Comment (yhwang) [reply .github/workflows/kfp-tekton-unittests.yml]: added `go.mod` now

Comment (yhwang) [reply .github/workflows/kfp-tekton-unittests.yml]: unfortunately, I can't find a good GitHub action to do change filtering. still need to code the filtering logic

**Comment by Tomcli** [comment]
thanks @yhwang 
/lgtm

**Pull request merged**